src/cell.cpp

```diff
@@ -1220,10 +1220,185 @@ extern "C" int openmc_cell_set_name(int32_t index, const char* name)
+          lattice->offset(distribcell_index, parent_cell.lattice_index);
+      }
+    }
+    return instance;
+  }
+
+  // Accessors
+  vector<ParentCell>& parent_cells() { return parent_cells_; }
+  const vector<ParentCell>& parent_cells() const { return parent_cells_; }
+
+  // Data Members
+  vector<ParentCell> parent_cells_;
+  std::unordered_map<int32_t, std::unordered_set<ParentCell, ParentCellHash>>
+    visited_cells_;
+};
+
+vector<ParentCell> Cell::find_parent_cells(int32_t instance) const
+{
+  ParentCellStack stack;
+  // start with this cell's universe
+  int32_t prev_univ_idx;
+  int32_t univ_idx = this->universe_;
+
+  while (true) {
+    const auto& univ = model::universes[univ_idx];
+    prev_univ_idx = univ_idx;
+
+    // search for a cell that is filled w/ this universe
+    for (const auto& cell : model::cells) {
+      // if this is a material-filled cell, move on
+      if (cell->type_ == Fill::MATERIAL) continue;
+
+      if (cell->type_ == Fill::UNIVERSE) {
+        // if this is in the set of cells previously visited for this universe, move on
+        if (stack.visited(univ_idx, {model::cell_map[cell->id_], C_NONE})) continue;
+
+        // if this cell contains the universe we're searching for, add it to the stack
+        if(cell->fill_ == univ_idx) {
+          stack.push(univ_idx, {model::cell_map[cell->id_], C_NONE});
+          univ_idx = cell->universe_;
+        }
+      } else if (cell->type_ == Fill::LATTICE) {
+        // retrieve the lattice and lattice universes
+        const auto& lattice = model::lattices[cell->fill_];
+        const auto& lattice_univs = lattice->universes_;
+
+        // start search for universe
+        auto lat_it = lattice_univs.begin();
+        while (true) {
+          // find the next lattice cell with this universe
+          lat_it = std::find(lat_it, lattice_univs.end(), univ_idx);
+          if (lat_it == lattice_univs.end()) break;
+
+          int lattice_idx = lat_it - lattice_univs.begin();
+
+          // move iterator forward one to avoid finding the same entry
+          lat_it++;
+          if (stack.visited(univ_idx, {model::cell_map[cell->id_], lattice_idx})) continue;
+
+          // add this cell and lattice index to the stack and exit loop
+          stack.push(univ_idx, {model::cell_map[cell->id_], lattice_idx});
+          univ_idx = cell->universe_;
+          break;
+        }
+      }
+      // if we've updated the universe, break
+      if (prev_univ_idx != univ_idx) break;
+    } // end cell loop search for universe
+
+    // if we're at the top of the geometry and the instance matches, we're done
+    if (univ_idx == model::root_universe && stack.compute_instance(this->distribcell_index_) == instance) break;
+
+    // if there is no match on the original cell's universe, report an error
+    if (univ_idx == this->universe_) {
+      fatal_error(fmt::format("Could not find the parent cells for cell {}, instance {}.", this->id_, instance));
```

Comment:
Once again, looks like this file needs to be clang-formatted

Reply:
Sorry about that. I've followed the instructions for installing the commit-hook that applies the style guide, but maybe I messed something up. 

```shell
All code changes were properly formatted
[find_parent_cells 1c1c848b3b4] Removing unused parameter in Cell::find_parent_cells.
 1 file changed, 27 insertions(+), 13 deletions(-)
```

I'll look into it.